x-pack/plugin/esql/qa/testFixtures/src/main/resources/rerank.csv-spec

```diff
@@ -0,0 +1,95 @@
+9032            | War and Peace: A Novel (6 Volumes)                    | Tolstoy Leo                                 | 0.02083333395421505
+2776            | The Devil and Other Stories (Oxford World's Classics) | Leo Tolstoy                                 | 0.01515151560306549
+;
+
+
+reranker using multiple fields
+required_capability: rerank
+required_capability: match_function
+
+FROM books METADATA _score
+| WHERE title:"war and peace" AND author:"Tolstoy"
+| RERANK "war and peace" ON title, author WITH "test_reranker"
+| KEEP book_no, title, author, _score
+;
+
+book_no:keyword | title:text                                            | author:text                                 | _score:double
+5327            | War and Peace                                         | Leo Tolstoy                                 | 0.02083333395421505
+9032            | War and Peace: A Novel (6 Volumes)                    | Tolstoy Leo                                 | 0.014285714365541935
+2776            | The Devil and Other Stories (Oxford World's Classics) | Leo Tolstoy                                 | 0.011363636702299118
+4536            | War and Peace (Signet Classics)                       | [John Hockenberry, Leo Tolstoy, Pat Conroy] | 0.009523809887468815
+;
+
+
+reranker after a limit
+required_capability: rerank
+required_capability: match_function
+
+FROM books METADATA _score
+| WHERE title:"war and peace" AND author:"Tolstoy"
+| SORT _score DESC
+| LIMIT 3
+| RERANK "war and peace" ON title WITH "test_reranker"
+| KEEP book_no, title, author, _score
+;
+
+book_no:keyword | title:text                         | author:text                                 | _score:double
+5327            | War and Peace                      | Leo Tolstoy                                 | 0.03846153989434242
+4536            | War and Peace (Signet Classics)    | [John Hockenberry, Leo Tolstoy, Pat Conroy] | 0.02222222276031971
+9032            | War and Peace: A Novel (6 Volumes) | Tolstoy Leo                                 | 0.02083333395421505
+;
+
+
+reranker before a limit
+required_capability: rerank
+required_capability: match_function
+
+FROM books METADATA _score
+| WHERE title:"war and peace" AND author:"Tolstoy"
+| RERANK "war and peace" ON title WITH "test_reranker"
+| KEEP book_no, title, author, _score
+| LIMIT 3
+;
+
+book_no:keyword | title:text                         | author:text                                 | _score:double
+5327            | War and Peace                      | Leo Tolstoy                                 | 0.03846153989434242
+4536            | War and Peace (Signet Classics)    | [John Hockenberry, Leo Tolstoy, Pat Conroy] | 0.02222222276031971
+9032            | War and Peace: A Novel (6 Volumes) | Tolstoy Leo                                 | 0.02083333395421505
+;
+
+
+reranker add the _score column when missing
+required_capability: rerank
+required_capability: match_function
+
+FROM books
```

Comment:
ℹ️ Added a case where the `_score` fiels is missing and check if it is added to the result